src/m3ninx/index/segment/fst/encoding/docs/slice.go

```diff
@@ -72,7 +68,7 @@ func (r *SliceReader) Doc(id postings.ID) (doc.Document, error) {
 }
 
 // AllDocs returns a docs iterator.
-func (r *SliceReader) AllDocs() index.IDDocIterator {
+func (r *SliceReader) Iter() index.IDDocIterator {
 	postingsIter := postings.NewRangeIterator(r.offset,
```

Comment:
We should get rid of the `r.offset` here too, only ever set to `0` here: https://github.com/m3db/m3/blob/master/src/dbnode/storage/index/compaction/compactor.go#L278

In fact, after changing this to be `0`, it seems that we can remove the `offset` passed into `postings.NewRangeIterator` entirely

Reply:
Agreed 👍